packages/javascript/bh-shared-ui/src/components/SSOProviderTable/SSOProviderTable.tsx

```diff
@@ -0,0 +1,195 @@
+        }}
+        {...props}
+    />
+));
+
+const SSOProviderTableActionsMenu: React.FC<{
+    onDeleteSSOProvider: () => void;
+}> = ({ onDeleteSSOProvider }) => {
+    /* Hooks */
+
+    const [anchorEl, setAnchorEl] = React.useState<HTMLElement | null>(null);
+
+    /* Event Handlers */
+
+    const handleOnOpen: React.MouseEventHandler<HTMLButtonElement> = (event) => {
+        setAnchorEl(event.currentTarget);
+    };
+
+    /* Implementation */
+
+    const onClickDeleteSSOProvider = useCallback(() => {
+        onDeleteSSOProvider();
+        setAnchorEl(null);
+    }, []);
+
+    return (
+        <>
+            <IconButton onClick={handleOnOpen} size='small'>
+                <FontAwesomeIcon icon={faEllipsisVertical} />
+            </IconButton>
+            <StyledMenu
+                anchorEl={anchorEl}
+                keepMounted
+                open={Boolean(anchorEl)}
+                onClose={() => {
+                    setAnchorEl(null);
+                }}>
+                <MenuItem onClick={onClickDeleteSSOProvider}>
+                    <ListItemIcon>
+                        <FontAwesomeIcon icon={faTrash} />
+                    </ListItemIcon>
+                    <ListItemText primary='Delete SSO Provider' />
+                </MenuItem>
+            </StyledMenu>
+        </>
+    );
+};
+
+const SSOProviderTable: React.FC<{
+    ssoProviders: SSOProvider[];
+    loading: boolean;
+    onDeleteSSOProvider: (ssoProviderId: SSOProvider['id']) => void;
+    onClickSSOProvider: (ssoProviderId: SSOProvider['id']) => void;
+    onToggleTypeSortOrder: () => void;
+    typeSortOrder?: 'asc' | 'desc';
+}> = ({ ssoProviders, loading, onDeleteSSOProvider, onClickSSOProvider, typeSortOrder, onToggleTypeSortOrder }) => {
+    const theme = useTheme();
+    return (
+        <Paper>
+            <TableContainer sx={{ maxHeight: 777 }}>
+                <Table stickyHeader aria-label='sso provider table'>
+                    <TableHead>
+                        <TableRow>
+                            <TableCell sx={{ backgroundColor: theme.palette.background.paper }} />
+                            <TableCell
+                                sx={{
+                                    fontWeight: 'bold',
+                                    verticalAlign: 'bottom',
+                                    backgroundColor: theme.palette.background.paper,
+                                }}>
+                                Provider Name
+                            </TableCell>
+                            <TableCell sx={{ fontWeight: 'bold', backgroundColor: theme.palette.background.paper }}>
+                                <TableSortLabel
+                                    active={!!typeSortOrder}
+                                    direction={typeSortOrder}
+                                    onClick={onToggleTypeSortOrder}>
+                                    Type
+                                </TableSortLabel>
```

Comment:
TIL about MUI `TableSortLabel`, thank yoU!

Nit - Can we disable/prevent this sorting functionality while loading or `ssoProviders.lenght === 0`? IMO The sorting should be a no op without anything to sort and so we shouldn't display a control for it until theres something to sort. 

Naively:
```js
{ssoProviders.length && !loading ? <TableSortLabel> : "Type"}
```

Reply:
Probably more like `ssoProviders.length > 1` since sorting only 1 entry also doesn't make any sense.